### PR TITLE
JIT: Fall back to a canonical SIMD handle in gtGetStructHandleForSimdOrHW

### LINKED ISSUE #80239: JIT: Invalid isSimdAsHWIntrinsic param passed in lclmorph
Run the following program on arm64 altjit with `DOTNET_JitNoStructPromotion=1`:
```csharp
using System.Numerics;
using System.Runtime.CompilerServices;

class Program
{
    static unsafe int Main(string[] args)
    {
        Unsafe.SkipInit(out Vector3 test);
        test.X = 500.0f;
        Consume(test);
        return 0;
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    private static void Consume(Vector3 test)
    {
    }
}
```

Output:
```scala
Assert failure(PID 48604 [0x0000bddc], Thread: 27132 [0x69fc]): Assertion failed 'varDsc->lvExactSize == 12' in 'Program:Main(System.String[]):int' during 'Lowering nodeinfo' (IL size 27; hash 0x38891b70; FullOpts)

    File: C:\dev\dotnet\runtime4\src\coreclr\jit\compiler.h Line: 3450
    Image: C:\dev\dotnet\runtime4\artifacts\bin\coreclr\windows.x64.Checked\CoreRun.exe
```

We have a tree:
```scala
fgMorphTree BB01, STMT00002 (before)
               [000007] --CXG------                         ▌  CALL      void   Program:Consume(System.Numerics.Vector3)
               [000014] ----------- arg0                    └──▌  HWINTRINSIC simd12 float Insert
               [000012] -----------                            ├──▌  LCL_VAR   simd12<System.Numerics.Vector3> V01 loc0         
               [000013] -----------                            ├──▌  CNS_INT   int    0
               [000003] -----------                            └──▌  CNS_DBL   float  500.00000000000000
```
and we try to create a local for it as part of passing the argument but then fail to obtain a struct handle to mark the new local with.

### PR DESCRIPTION
**Pull request by jakobbotsch**
Fix #80239

### CODE REVIEW

File: src/coreclr/jit/compiler.h [hunk 1]
```diff
@@ -8449,6 +8449,11 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
             clsHnd = gtGetStructHandleForHWSIMD(simdType, simdBaseJitType);
         }
 
+        if (clsHnd == NO_CLASS_HANDLE)
+        {
+            clsHnd = gtGetCanonicalStructHandleForSIMD(simdType);
```
Comment (tannergooding) [inline src/coreclr/jit/compiler.h:8454]: Do we want/need a comment explaining why the fallback is here?

**Review by tannergooding — APPROVED** [review]

File: src/coreclr/jit/compiler.h (hunk 1 above)
Comment (jakobbotsch) [reply src/coreclr/jit/compiler.h:8457]: Added a comment, does it look good?

Comment (tannergooding) [reply src/coreclr/jit/compiler.h:8457]: LGTM

**Comment by tannergooding** [comment]
@jakobbotsch looks like some path is passing in a non simdType and that's causing an `unreached` to be hit.

**Comment by jakobbotsch** [comment]
We call `gtGetStructHandleForSimdOrHW` on
```scala
N002 (  2,  2)              [000001] ---XG------ >>>                     ▌  HWINTRINSIC struct int LoadPairScalarVector64 <l:$183, c:$182>
N001 (  1,  1)              [000000] -----------                         └──▌  LCL_VAR   long   V00 arg0         u:1 (last use) $80
```
from `gtGetStructHandleIfPresent`. I'll make `gtGetStructHandleIfPresent` return `NO_CLASS_HANDLE` for non-SIMD HW intrinsics.

File: src/coreclr/jit/compiler.h [hunk 2]
```diff
@@ -8438,6 +8438,8 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
                                                       CorInfoType simdBaseJitType,
                                                       bool        isSimdAsHWIntrinsic = false)
     {
+        assert(varTypeIsSIMD(simdType));
```
Comment (tannergooding) [inline src/coreclr/jit/compiler.h:8441]: Should we likewise add an assert to `gtGetStructHandleForSIMD` and `gtGetStructHandleForHWSIMD`?

Comment (jakobbotsch) [reply src/coreclr/jit/compiler.h:8443]: Good idea, added

**Review by tannergooding — APPROVED** [review]

**Pull request merged**